### PR TITLE
refactor: Introduce chart wrapper

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

This refactoring is made to support the fit-height feature (https://github.com/cloudscape-design/components/pull/1273) that is to be added for all charts.

### How has this been tested?

Existing tests. Screenshot tests

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/mixed-line-bar-chart/internal.tsx [hunk 1]
```diff
@@ -204,28 +203,37 @@ export default function InternalMixedLineBarChart<T extends number | string | Da
   const reserveFilterSpace = !showChart && !isNoMatch && (!hideFilter || additionalFilters);
   const mergedRef = useMergeRefs(containerRef, __internalRootRef);
 
+  const filterItems: ChartFilterProps<MixedLineBarChartProps.ChartSeries<T>>['series'] = useMemo(
```
Comment (timogasda) [inline src/mixed-line-bar-chart/internal.tsx:206]: nit: I'm not sure we really need a memo for a simple `map` function that usually processes 1-10 items

File: src/pie-chart/pie-chart.tsx [hunk 2]
```diff
@@ -22,6 +19,7 @@ import ChartPlot, { ChartPlotRef } from '../internal/components/chart-plot';
 import { SomeRequired } from '../internal/types';
 import { useInternalI18n } from '../internal/i18n/context';
 import { nodeBelongs } from '../internal/utils/node-belongs';
+import { PieArcDatum } from 'd3-shape';
```
Comment (timogasda) [inline src/pie-chart/pie-chart.tsx:22]: nit: can we move this import up?

**Review by timogasda — APPROVED** [review]
Looks great! Just two small comments that you can ignore or address later

File: src/mixed-line-bar-chart/internal.tsx (hunk 1 above)
Comment (pan-kot) [reply src/mixed-line-bar-chart/internal.tsx:206]: Thanks, I copied it from the existing chart-filters.tsx actually and forgot to remove the original file. I'll remove the file and get rid of the unnecessary memo.

**Review by timogasda — APPROVED** [review]

**Pull request merged**